packages/frontapp/src/app/components/Dropdown/index.tsx

```diff
@@ -131,13 +137,18 @@ const Dropdown: FC<IDropdownProps> = ({
   useOnClickOutside(containerRef, () => {
     setShowOptionList(false);
   });
-  const [selectedOptionIndex, setSelectedOptionIndex] =
-    useState<number>(initIndex);
+  const [selectedOptionIndex, setSelectedOptionIndex] = useState<number>(() => {
+    let optionIndex = -1;
+    if (value !== undefined) {
+      optionIndex = options.findIndex((d) => d.value === value);
+    }
+    return optionIndex > -1 ? optionIndex : initIndex;
```

Comment:
it might be better to put this logic into a separate function or variable?

Reply:
Could you please explain a bit more?